### PR TITLE
Move OpenStack periodics from release to shiftstack

### PR DESCRIPTION
**Pull request by EmilienM**
Moving the jobs from release to shiftstack, so the OpenStack team own
them.

### CODE REVIEW

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-release-4.10.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    cluster_profile: openstack-vexxhost
+    env:
+      FIPS_ENABLED: "true"
+      TEST_SUITE: experimental/reliability/minimal
+    pre:
+    - chain: ipi-openstack-pre
+    - ref: fips-check
+    workflow: openshift-e2e-openstack-ipi
+zz_generated_metadata:
+  branch: release-4.10
```
Comment (EmilienM) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-release-4.10.yaml:38]: the branch doesn't exist on that repo, I think this will fail. I'll probably create one for testing.

Comment (petr-muller) [reply ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-release-4.10.yaml:38]: What code do you want to test? It may make more sense to create a "variant" config, with a file called like `shiftstack-shiftstack-ci-master__4.10.yaml` which means "testing code from master but somehow this is a different config than we have for master branch itself".

**Comment by EmilienM** [comment]
/hold
this seems to be working, I'll move all other periodics.

**Comment by mdbooth** [comment]
To verify that the changes in `ci-operator/jobs` were automatically generated, I reverted them and re-ran `make update`. There was  a remaining diff in `ci-operator/jobs/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main-periodics.yaml`. The following fragment appears to have been manually added to jobs in that file:

```yaml
   reporter_config:
     slack:
       channel: '#shiftstack-bot'
       job_states_to_report:
       - failure
       - error
       report_template: ':volcano: Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
         <{{.Status.URL}}|View logs> :volcano:'
```

Is this correct?

**Review by mdbooth — CHANGES_REQUESTED** [review]
We also need to ensure we fully understand the organisational impact of moving this stuff.

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.10-upgrade-from-stable-4.9.yaml [hunk 2]
```diff
@@ -1,8 +1,4 @@
 base_images:
-  openstack-installer:
-    name: "4.9"
-    namespace: ocp
-    tag: openstack-installer
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.10-upgrade-from-stable-4.9.yaml:5]: but since there is no more OpenStack job in this file, it should be fine

[the remark this answers is not in the document]
Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.10-upgrade-from-stable-4.9.yaml:5]: Indeed it should be fine, as the `openstack-installer` image is only used in OpenStack-specific steps in the step registry:

```
$ rg -l openstack-installer ci-operator/step-registry | xargs -n1 dirname | sort
ci-operator/step-registry/ipi/conf/openstack/precheck
ci-operator/step-registry/openstack/conf/clouds
ci-operator/step-registry/openstack/conf/creatednsrecords
ci-operator/step-registry/openstack/conf/createfips
ci-operator/step-registry/openstack/conf/exportcindercredentials
ci-operator/step-registry/openstack/conf/generateconfig
ci-operator/step-registry/openstack/conf/resourcenames
ci-operator/step-registry/openstack/deprovision/bastionproxy
ci-operator/step-registry/openstack/deprovision/deletednsrecords
ci-operator/step-registry/openstack/deprovision/deletefips
ci-operator/step-registry/openstack/deprovision/machinesubnet
ci-operator/step-registry/openstack/gather
ci-operator/step-registry/openstack/noop
ci-operator/step-registry/openstack/provision/bastionproxy
ci-operator/step-registry/openstack/provision/machinesubnet
ci-operator/step-registry/openstack/provision/proxyconf
ci-operator/step-registry/openstack/test/manifests
ci-operator/step-registry/shiftstack/cleanup
ci-operator/step-registry/shiftstack/liveliness
ci-operator/step-registry/shiftstack/monitor
ci-operator/step-registry/shiftstack/monitor/expose
ci-operator/step-registry/shiftstack/quotas
```

[the remark this answers is not in the document]
Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.10-upgrade-from-stable-4.9.yaml:5]: and no job workflow other than ours seems to refer to openstack steps:

```
$ rg -l 'openstack' ci-operator/step-registry/openshift/e2e | sort
ci-operator/step-registry/openshift/e2e/openstack/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/byon/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/byon/openshift-e2e-openstack-byon-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/byon/openshift-e2e-openstack-byon-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/ccm/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/ccm/install/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/ccm/install/openshift-e2e-openstack-ccm-install-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/ccm/install/openshift-e2e-openstack-ccm-install-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/ccm/openshift-e2e-openstack-ccm-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/ccm/openshift-e2e-openstack-ccm-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/csi/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/csi/cinder/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/csi/cinder/openshift-e2e-openstack-csi-cinder-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/csi/cinder/openshift-e2e-openstack-csi-cinder-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/csi/manila/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/csi/manila/openshift-e2e-openstack-csi-manila-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/csi/manila/openshift-e2e-openstack-csi-manila-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/nfv/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/nfv/openshift-e2e-openstack-nfv-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/nfv/openshift-e2e-openstack-nfv-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/ovn/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/ovn/openshift-e2e-openstack-ovn-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/ovn/openshift-e2e-openstack-ovn-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/proxy/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/proxy/openshift-e2e-openstack-proxy-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/proxy/openshift-e2e-openstack-proxy-workflow.yaml
ci-operator/step-registry/openshift/e2e/openstack/techpreview/OWNERS
ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.metadata.json
ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.yaml
ci-operator/step-registry/openshift/e2e/ovirt/minimal/test/openshift-e2e-ovirt-minimal-test-commands.sh
ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```

**Comment by EmilienM** [comment]
openstack 4.9 jobs failing for a known reason: Bug 2027329
therefore they can be skipped.

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 3]
```diff
@@ -1312,17 +1312,6 @@ periodics:
     job-release: "4.10"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-release-master-ci-4.10-e2e-aws-upgrade-single-node
-  reporter_config:
```
Comment (alvaroaleman) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:1315]: Why are yo removing the slack reporting for single node ci? This seems unrelated to openstack

Comment (EmilienM) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:1315]: weird. I just ran `make update`.

Let me try again.

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.8-upgrade-from-stable-4.7.yaml [hunk 4]
```diff
@@ -1,12 +1,4 @@
 base_images:
-  openstack-installer:
-    name: "4.6"
```
Comment (mandre) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.8-upgrade-from-stable-4.7.yaml:3]: Should be "4.7"

**Review by mandre — COMMENTED** [review]
I rebased the commit and re-generated `ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml` and `ci-operator/jobs/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main-periodics.yaml` with `make jobs`.

Apart from moving the periodic jobs under shiftstack, this also consolidates all of our periodics to run from the `nightly` image stream which is I think what we want.

/lgtm
